go/vt/vtgate/planbuilder/testdata/filter_cases.json

```diff
@@ -846,6 +846,44 @@
       ]
     }
   },
+  {
+    "comment": "Merging subqueries should remove keyspace from query",
+    "query": "select u.id from user.user as u where not exists (select 1 from user.user_extra as ue where u.id = ue.user_id)",
+    "v3-plan": {
+      "QueryType": "SELECT",
+      "Original": "select u.id from user.user as u where not exists (select 1 from user.user_extra as ue where u.id = ue.user_id)",
+      "Instructions": {
+        "OperatorType": "Route",
+        "Variant": "Scatter",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
```

Comment:
add a test case when the subquery has a system table in it. That database qualifier should not be removed.